hardware_interface/src/resource_manager.cpp

```diff
@@ -195,6 +197,58 @@ class ResourceStorage
     return result;
   }
 
+  void remove_all_hardware_interfaces_from_available_list(const std::string & hardware_name)
+  {
+    // remove all command interfaces from available list
+    for (const auto & interface : hardware_info_map_[hardware_name].command_interfaces)
+    {
+      auto found_it = std::find(
+        available_command_interfaces_.begin(), available_command_interfaces_.end(), interface);
+
+      if (found_it != available_command_interfaces_.end())
+      {
+        available_command_interfaces_.erase(found_it);
+        RCUTILS_LOG_DEBUG_NAMED(
+          "resource_manager", "(hardware '%s'): '%s' command interface removed from available list",
+          hardware_name.c_str(), interface.c_str());
+      }
+      else
+      {
+        // TODO(destogl): do here error management if interfaces are only partially added into
+        // "available" list - this should never be the case!
+        RCUTILS_LOG_WARN_NAMED(
+          "resource_manager",
+          "(hardware '%s'): '%s' command interface not in available list."
+          " This can happen due to multiple calls to 'cleanup'",
```

Comment:
```suggestion
          " This should not happen (hint: multiple cleanup calls).",
```
How about this? I think the multiple cleanup calls case can only happen in tests, not really through CM interfaces so the print should be primarily communicating to the end user.

Perhaps we don't even need to print the hint but leave ideas here for debugging in a comment.

Reply:
What would be an idea to debug? I think it is easier to describe behavior than to try to figure out when this could happen.

Reply:
What I mean is that you warm about this case happening after multiple Clea up calls but that is only really happening in unit tests and not during normal operation w CM.